model/src/main/proto/performance_metrics.proto

```diff
@@ -0,0 +1,149 @@
+  message ApkSize {
+    // Size of the android package file of the app on the device in bytes.
+    int64 apk_size_bytes = 1;
+  }
+
+  // Structure of the storage usage package.
+  message StorageUsage {
+    // Amount of storage usage by the app on user's device in bytes.
+    int64 storage_usage_bytes = 1;
+  }
+
+  // Structure of the startup latency package.
+  message StartupLatency {
+    // Amount of latency on application startup in millis.
+    int64 startup_latency_millis = 1;
+  }
+
+  // Structure of the memory usage package.
+  message MemoryUsage {
+    // Amount of memory used by the application on the device in bytes.
+    int64 total_pss_bytes = 1;
+  }
+
+  // Structure of the network usage package.
+  message NetworkUsage{
+    // Number of bytes received by the application on the device during the bucket's time interval.
+    int64 bytes_received = 1;
+
+    // Number of bytes transmitted by the application from the device during the bucket's time
+    // interval.
+    int64 bytes_transmitted = 2;
+  }
+
+  // Structure of the cpu usage package.
+  message CpuUsage{
+    // Amount of cpu used by the application.
+    int64 cpu_usage = 1;
+  }
+
+  // Supported priority of events for performance metric logging.
+  enum Priority {
+    // The undefined priority of an event.
+    PRIORITY_UNSPECIFIED = 0;
+
+    // The priority of events whose logs will be the first to be removed from the storage if the
+    // size exceeds a certain limit
+    LOW_PRIORITY = 1;
+
+    // The priority of events whose logs will be the ones to be removed from the storage if the
+    // size exceeds a certain limit and no low priority logs exist.
+    MEDIUM_PRIORITY = 2;
+
+    // The priority of events whose logs should not be removed from the storage on most occasions
+    // but can be removed if they're the only ones there and size limit exceeds.
+    HIGH_PRIORITY = 3;
+  }
+
+  // Supported activities for indicating the current activity where the event logging is occuring.
+  enum CurrentActivity {
+    // The undefined current activity of an event.
+    ACTIVITY_UNSPECIFIED = 0;
+
+    // The metric log was taken when home activity was opened.
+    HOME_ACTIVITY = 1;
+    // TODO(#4340): Add support for all activities which are going to be used for metric logging.
+  }
+
+  // Supported memory tiers for indicating the total memory of user's device.
+  enum MemoryTier {
+    // The undefined memory tier of a device.
+    MEMORY_TIER_UNSPECIFIED = 0;
+
+    // The memory tier where the amount of memory is less than or equal to 1GB.
+    LOW_MEMORY_TIER = 1; // =<1GB
+
+    // The memory tier where the amount of memory is less than or equal to 2GB and more than 1GB.
+    MEDIUM_MEMORY_TIER = 2; // =<2GB && >1GB
+
+    // The memory tier where the amount of memory is greater than 2GB.
+    HIGH_MEMORY_TIER = 3; // >2GB
+  }
+
+  // Supported storage tiers for indicating the total storage space on user's device.
```

Comment:
Is there any external reference documentation that we can point to that supports the size designations for each of these size tiers?

Ditto for MemoryTier.

Reply:
There is no clear-cut documentation that supports the size designations for each of these size tiers. It was something that we came up with during our initial discussions and thought was apt. Do you have any reference doc suggestions regarding this?

Reply:
I don't, but perhaps we should be doing a brief investigation of the distribution of available storage on Android devices throughout the world to get an idea on what these tiers should be (versus arbitrarily setting them).

Reply:
I did a brief research on this topic and unfortunately, I couldn't find anything supporting the classification of android devices on the basis of storage/memory space. Perhaps it is because of the rapidly changing storage and memory space configurations in newer android devices. It'll be really helpful if you could suggest something here :)

Reply:
Taking into account the data Isalo provided, I've modified the memory and storage tiers to be somewhat like:  

1. Memory Tiers 
- Low --> <2GB
- Medium --> <=3GB && >=2GB
- High --> >3GB

2. Storage Tiers 
- Low --> <32GB
- Medium --> <=64GB && >=32GB
- High --> >64GB

I've taken the device market share + device model specs (memory and storage) into account to come up with this. Does this suffice?

Reply:
Yup this seems quite sensible--thanks!